lock/etcd.go

```diff
@@ -74,16 +75,18 @@ func NewEtcdLockClient(keyapi KeysAPI, group string) (*EtcdLockClient, error) {
 // So we first try to get the value, if the value is not found we create the key
 // with a default semaphore value
 func (c *EtcdLockClient) Init() error {
-	if _, err := c.Get(); err != nil {
-		if errors.Is(err, ErrNotFound) {
-			sem := newSemaphore()
-			if err := c.Set(sem); err != nil {
-				return fmt.Errorf("unable to set semaphore: %w", err)
-			}
-
-			return nil
-		}
-		return fmt.Errorf("unable to get semaphore: %w", err)
+	sem := newSemaphore()
+	payload, err := json.Marshal(sem)
+	if err != nil {
+		return fmt.Errorf("unable to marshal initial semaphore: %w", err)
+	}
+
+	if _, err := c.keyapi.Txn(context.Background()).If(
```

Comment:
Given that this is not a top-level context, I think using `context.TODO()` is more appropriate here.
```suggestion
	if _, err := c.keyapi.Txn(context.Background()).If(
```